### ISSUE TITLE
💗 the time you clearly spent polishing your application

### ISSUE THREAD

**Issue by PySimpleGUI** [dennisbyington/wordle_gui#1]
Your window is gorgeous!

![image](https://user-images.githubusercontent.com/46163555/186405930-47b37355-7256-46a6-9190-3ae5644ff6e4.png)

And I loved seeing your use of tuples for keys. ![image](https://user-images.githubusercontent.com/46163555/186406168-c086596c-9239-4702-89f1-cc1d9e147a81.png)

GREAT way to start my day.

I wasn't able to see the results on my screen however.  The letters didn't show up until I accidently highlighted a square and then then it disappeared quickly.  Maybe I simply don't know how it works.

A screenshot of your fantastic window in your readme will grab attention of visitors for SURE.

I see you made use of JSON.  Did you know about the simplification of JSON files that PySimpleGUI provides?  I'm biased of course and use it quite a bit.... again, out of laziness since there's no reading, writing, etc.  And accessing them looks like a dictionary lookup.,  I'm guessing you've likely hears about them already, but thought I would toss them out for ya just in case.

I always get excited seeing games made with PySimpleGUI.  Here's my somewhat poor attempt at Worlde

https://pysimplegui.trinket.io/demo-programs#/games/wordle

If you have a moment to post  screenshot over in Issue 10 of the PySimpleGUI GitHub, I am sure users would love to see your creation!  It's inspiring.

**Comment by dennisbyington** [comment]
Thank you for the comments and pointing out the bugs!  I believe the invisible letters is due to me working on a Mac and not considering Windows machines and their default systems settings.  I have updated the font colors to white which should fix that issue.  

I found PySimpleGUI after attempting a few other GUI libraries and immediately saw that PSG was far easier to understand and to get implemented, so thank you for building a great product.  It was your Wordle demo that I found the idea of tuples as keys, so again thank you for the inspiration. 

So far I have only used the dump and load json functions, but I plan to dive deeper into the json functionality as I dig deeper into PSG.  I will post a screenshot into the PSG GitHub and will work on updating my readme files.

Thank you again!

**Comment by PySimpleGUI** [comment]
![image](https://user-images.githubusercontent.com/46163555/186515415-a05bd731-7038-429c-8f3b-0725fd1d7348.png)

OMG I got one right!

I think this is the first time I've won a game.  Clearly I don't play often, but still, I won one!  LOL

I saw your screenshot over in the PySimpleGUI Repo.  THANK YOU for posting it!

How about getting one into your repo's readme?

In case you don't know the process, I'll step you through it.

First, let's copy that screenshot from Issue 10 to here:

![image](https://user-images.githubusercontent.com/46163555/186516232-d41b1f21-9312-4fba-8969-cc5ff9efa8f8.png)

Oooo... notice you're app is running successfully on both the Mac and Windows.  Very nice.

1. Rename your readme.txt to be readme.md - you're going to use the markdown format 👍  It's a fantastic skill for you to learn and seeing this application of yours tells me  learning markdown is going to be trivial for you
2. Remove the lines that look like this:
  `# ---------------------------------------------------------------------------`
3. Notice how the other lines where you used a "#" that they turned into headers?  That's your first lesson in Markdown... the second lesson is coming up next, adding images.  Oh!  and check out the bulleted lists that you've already made too.  You're very close to being done with a really nice readme.
4. Choose the location you want to put your screenshot in your readme
5. Copy and paste this line into your readme at that spot:
`![image](https://user-images.githubusercontent.com/46163555/186516232-d41b1f21-9312-4fba-8969-cc5ff9efa8f8.png)`

So, where did I get that line of text?  I got it from writing this GitHub issue comment.  In the PySimpleGUI Cookbook I show how you can use GitHub issues to store screenshots using this trick.  All I did was paste then image (a literal copy and paste of the image) and GitHub uploaded it and turned it into a piece of markdown code (the line of text).                                

You've made an impressive program, with a GUI, it's worth sharing with everyone.  In the future  you'll learn more about markdown.

----------------------

**This is your readme.txt file copied into this issue.  Because the GitHub comments are in Markdown format,  you'll get a preview of what it'll look like by looking at it here....**
--------------------



# Background

This integrates much of the logic in my wordle_cli program and was done specficially to learn a GUI library.


# ---------------------------------------------------------------------------

# Description 

GUI implementation of the Wordle game 

  
# ---------------------------------------------------------------------------
# Game introduction:

Wordle is a word guessing game.  Words are restricted to five letters and players get six guesses.  If a letter is in the correct spot in the answer, it will turn green.  If it is in the answer, but in the wrong spot it will turn yellow.  If it is not in the answer it will turn gray.  Multiples of the same letter (such as the o's in color) will be colored green or yellow as appropriate if the answer also contains multiple instances of the letter, otherwise the duplicates will be gray.  

Additional information can be found here: https://en.wikipedia.org/wiki/Wordle


# ---------------------------------------------------------------------------
# Dependencies

- Python3 
- Imports: argparse, sys, json, datetime
- External imports: matplotlib, PySimpleGUI

* Install the external requirments via terminal: 
    'pip install -r requirements.txt'

# ---------------------------------------------------------------------------
# Help
  
Help can be obtained by including the [-h] option to the program:

  ./wordle_cli.py -h


# ---------------------------------------------------------------------------
# Author

Dennis Byington
dennisbyington@mac.com


# ---------------------------------------------------------------------------
# Version history & release notes

0.1 - Initial release


# ---------------------------------------------------------------------------
# Bugs

- Deleting one letter deletes the entire word.  I am still thinking through the process of deleting one letter at a time.  

- I am also seeking inputs and constructive criticism on areas I can improve. 


# ---------------------------------------------------------------------------
# Future features

- None


# ---------------------------------------------------------------------------
# License

This software is licensed under the MIT License.  See license.txt for details.


# ---------------------------------------------------------------------------
# Acknowledgments

This is my sofware recreation the web-based Wordle game.  Wordle was originally created by Josh Wardle and is now owned and published by the New York Times.

**Comment by PySimpleGUI** [comment]
Oh!  And thank you for sharing how you found PySimpleGUI.  
![image](https://user-images.githubusercontent.com/46163555/186520151-280924c3-17e4-4914-889d-efd4a9e16fd3.png)


Hearing your story was not only helpful, but it's motivating.  You clearly, _**clearly**_, wanted to make a GUI and were determined enough to try several different librariares until you found that one that fit you and your project.  That's the kind of determination I can get behind and draw energy from!

**Comment by PySimpleGUI** [comment]
## JSON Files versus PySimpleGUI User Settings

I brought up the PySimpleGUI User Settings simply because I saw your application uses JSON files.

> So far I have only used the dump and load json functions, but I plan to dive deeper into the json functionality as I dig deeper into PSG.

The point of the User Settings API was to _**simplify**_ the access of JSON and CONFIG files.  The idea was to hide the fact you're using a JSON file at all and instead have your brain believing that you're simply manipulating a dictionary.

You'll find the feature documented in the main docs:

https://www.pysimplegui.org/en/latest/#user-settings-api

I noticed you've named your json files with .txt.  It's recommended that you use .json for json files.  IT doesn't matter either way though.

Here's an example of using the UserSerrtings object to access your jSON file that has the stats in it.

![image](https://user-images.githubusercontent.com/46163555/186534310-24336a9e-d474-48b8-b3aa-b1eaf40db04b.png)


Here's the code I used:
```python
import PySimpleGUI as sg

settings = sg.UserSettings(r'E:\DownloadsE\wordle_gui-master\wordle_gui-master\data_files\stats.txt')

sg.Print('Total Games Played = ', settings['tot_games_played'])
sg.Print('Win Percentage = ', settings['win_percent'])
sg.Print('Your entire JSON file as a dictionary:')
sg.Print(settings, colors='white on green')
sg.Print(wait=True)
```


The idea is to simplify your life by abstracting away the JSON file stuff.  You know how to use JSON files, this you've shown in your code.  PySimpleGUI is simply giving you the option of interfacing with them differently.

I'm a "lazy programmer" in many ways, and I found that it's easier to write a library to do something and then use that library instead of the original library.

I wanted to remove the need to worry about opening, writing, and closing the file and instead focus on messing around with the insides part.... the settings.

Y"our current of using jSON files is OK, it's fine, whatever you like is good enough.  You're getting the job done... that's the important thing.

**Comment by dennisbyington** [comment]
Glad it's working and you got a win!  This is one of my first programs to share online and I'm excited to see it working for other folks.

I was able to update the readme, thank you for all the help and making that a super easy process!  And the trick for storing the pictures was pretty neat.  

I see what you're saying about the JSON files now, I wasn't aware that you could load direct into a variable like that - I can see the benefits of abstracting away the file handling - I will definitely give it a try!  

PSG definitely fit me and my project - it was the only library that felt intuitive.  I was able to read through the documentation, look at the examples, and make sense of it immediately.  And it was super easy to up and running - it's a great resource!  

Thank you again for all of your help, you have been amazingly kind and instructive!

**Comment by PySimpleGUI** [comment]
So... Dennis... I'm _**fascinated**_ by the beauty of your window.

Have you done graphic design work in the past?  I understand thsi is your first Python GUI, but have you written others before?

It's very very rare that I see a window be produced, using standard easy to access PySimpleGUI features, that turns out so polished as yours did.,

The impressive features are:
* The matching color pallet
* The keyboard layout of letters at the bottom, AND they're multiple colors
* The relief around the letter of each row and the thickness of the border
* The score showed as a distribution

These are not the traits of a beginner.

**Comment by PySimpleGUI** [comment]
Or, as I've said in the documentation somewhere long ago
> "Beautiful windows don't just happen"

**Comment by dennisbyington** [comment]
Thank you, I did spend a decent amount of time tweaking those windows, so I'm really glad that someone besides myself appreciated them!  

This was my first gui, I had only drawn a few ideas out for a software engineering course I took but had never coded one.  I wanted to do a decent dive into GUIs so I took my time on this one and really read into the documentation so I could get a solid foundation of the function calls and the options available.  

I'm not artistic, so I wanted to recreate a UI I had seen before - I used the actual Wordle game (screenshots below) to emulate and I tried to recreate those screens as best as I could.  This turned out to be an excellent way to learn the different options available to format the window and objects.  I had used MATLAB before, so learning the Matplotlib library to graph the distribution was a fun rabbit hole to go down!  

![wordle_screengrab](https://user-images.githubusercontent.com/106843224/187097105-167d0cdc-eb87-4a76-9484-4b420236bc3f.png)

![Screen Shot 2022-08-28 at 5 58 01 PM](https://user-images.githubusercontent.com/106843224/187098003-5560f669-6dee-4cf7-b1ea-0aee6d81107e.png)
